noir-projects/noir-contracts/contracts/app/crowdfunding_contract/src/main.nr

```diff
@@ -110,4 +117,34 @@ pub contract Crowdfunding {
         emit_event_in_public(WithdrawalProcessed { amount, who: to }, &mut context);
     }
     // docs:end:operator-withdrawals
+
+    #[utility]
+    unconstrained fn get_donation_notes(
+        donor: AztecAddress,
+        page_index: u32,
+    ) -> BoundedVec<RetrievedNote<UintNote>, MAX_NOTES_PER_PAGE> {
+        let storage_slot = storage.donation_receipts.at(donor).get_storage_slot();
+
+        // We bypass Aztec.nr's higher-level abstractions here to access the metadata available only in RetrievedNote
+        // type that is not returned by view_notes function. We on the other hand need this metadata because we will
+        // be proving the note existence in the Claim contract.
```

Comment:
This might be interesting for @nventuro 

Also, this is a great example where having a utility function be callable from another contract would be valuable (the Claim contract is the destination where this note ends up hence would make sense to just call it from there).